docs/understanding-radicle/how-radicle-works.md

```diff
@@ -0,0 +1,316 @@
+}
+
+struct Project {
+    /// A short name
+    name: String,
+
+    /// A slightly longer description (should fit in a headline)
+    description: Option<String>,
+
+    /// The default branch. "master" is assumed for git repositories if
+    /// unspecified.
+    default_branch: Option<String>,
+}
+```
+
+An identity can also be extended with additional metadata as needed by the client or interface that's interacting with
+it.
+
+When you create a user identity, it's signed with an Ed25519 keypair that ensures only you can change the identity and
+push changes to the Radicle network on behalf of your peer ID.
+
+When you create a project, your identity is added as a **delegate**—more on those in a moment.
+
+### Why Radicle starts with identity
+
+At their core, Git repositories are just commit histories. Eventually, these commit histories develop into complex trees
+of folders, files, and code.
+
+But if you take two Git repositories at face value, there's no genuine way of comparing them. You can look at the commit
+logs and diff individual files, but there's no way of knowing, just through the data that is stored in Git, whether
+project A and project B are meant to converge someday. Are their histories and `HEAD` meant to be the same? Can you tell which repository is meant to be canonical?
+
+All this uncertainty happens because source code is *portable*—it can be cloned, edited, and hosted somewhere else
+without the project's maintainer knowing about it.
+
+Centralized code collaboration platforms like GitHub and GitLab create also create identities for the projects they
+host. If you create two identical repositories on these platforms, their identities are `github.com/bob/project` and
+`gitlab.com/bob/project`. You might know they're the same, but to someone else, there's no guarantee that the projects
+are the same despite seemingly sharing a `bob/project` identity.
+
+With centralized platforms, the code remains portable, but the *identity is non-portable*.
+
+A portable Radicle identity solves this problem because you always own your Radicle identity. Your identity might be replicated across the Radicle network, but because your key signs it, it's always yours. Your project may be replicated across the Radicle network, but because your identity is a delegate, only you can change the canonical remote/branch.
+
+Radicle identities are perfect for a peer-to-peer protocol because *the data is shipped with the source code, but not in
+the source code*.
+
+### Delegates
+
+Aside from is portable identity, the most important part of a project is the delegates.
+
+When you create a project, you add your identity as a delegate. Every time you push new changes to the project, they're
+signed with your Radicle identity and your key.
+
+When a project has multiple contributors, all of whom may be pushing code and patches to their individual remotes, the
+delegate's remote is considered the "canonical" version of the project's code base.
```

Comment:
This is still a bit unclear to me. If you have multiple delegates, they still all have their own branches, which may well be diverging. I think we might leave it up to projects to decide how they determine canonicity between multiple delegates. For example it might be just quorum on the most recent commit in a branch.

If we have a project with 2 delegates and their respective default branches `main` (commits from bottom to top, top being the newest):
```
commit 3                 commit 5
commit 2                 commit 2
commit 1                 commit 1
delegate A `main` |  delegate B `main`
```

Quorum here is at commit 2.

But there might also be other mechanisms that projects choose to determine what is the agreed upon latest state.

Reply:
Thanks for the information and clarity—I really like the explanation you gave here, so I went ahead and adapted it for the doc itself!

I wrote a large chunk of this section all the way back in March or April of this year, so I think a *lot* has changed around the way identity is implemented across the network.